homeassistant/components/backup/websocket.py

```diff
@@ -147,6 +149,38 @@ async def handle_restore(
         connection.send_result(msg["id"])
 
 
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "backup/can_decrypt_on_download",
+        vol.Required("backup_id"): str,
+        vol.Required("agent_id"): str,
+        vol.Required("password"): str,
+    }
+)
+@websocket_api.async_response
+async def handle_can_decrypt_on_download(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """Check if the supplied password is correct."""
+    try:
+        await hass.data[DATA_MANAGER].async_can_decrypt_on_download(
+            msg["backup_id"],
+            agent_id=msg["agent_id"],
+            password=msg.get("password"),
+        )
+    except IncorrectPassword:
+        connection.send_error(msg["id"], "password_incorrect", "Incorrect password")
+    except UnsuppertedSecureTarVersion:
```

Comment:
I think it's the manager the should catch these exceptions and in turn raise a `BackupManagerError`.

Reply:
There are other possible exceptions from the util that we don't catch here.